notes/connection-lifecycle.md

```diff
@@ -6,78 +6,125 @@
+)
 
-Set the connection's `timeout` property to the "connect timeout" value.
+conn.set_tunnel("example.com", scheme="http", headers={"Proxy-Header": "value"})
+```
 
 Connect to the first origin by calling the `HTTPConnection.connect()` method.
 If an error occurs here you can check whether the error occurred during the
-connection to the proxy if `HTTPConnection._connecting_to_proxy` is true.
+connection to the proxy if `HTTPConnection.has_connected_to_proxy` is true.
 If the value is false then the error didn't occur while connecting to a proxy.
 
-(Should we be setting `timeout` to some write timeout at this point? `None`?)
+```python
+# Explicitly connect to the origin. This isn't
+# required as sending the first request will
+# automatically connect if not done explicitly.
+conn.connect()
+```
 
-After connecting to the origin, the connection can be checked to see if `is_verified` is set to true. If not the pool emits a warning.
-The warning only matters for when verification is disabled, because otherwise an error is raised on unverified TLS handshake.
+After connecting to the origin, the connection can be checked to see if `is_verified` is set to true. If not the `HTTPConnectionPool` would emits a warning. The warning only matters for when verification is disabled, because otherwise an error is raised on unverified TLS handshake.
 
-Then the HTTP request can be sent with `HTTPConnection.request()`. If a `BrokenPipeError` (should we include `SSLEOFError`?)
-is raised while sending the request body it can be swallowed as a response
-can still be received from the origin even when the request isn't completely
-sent.
+```python
+if not conn.is_verified:
+    # There isn't a verified TLS connection to target origin.
+if not conn.is_proxy_verified:
+    # There isn't a verified TLS connection to proxy origin.
+```
 
-Set the connections `timeout` property is set to the `read_timeout`. (Currently this appears calculated via some algorithm?)
+If the read timeout is different from the connect timeout then the
+`HTTPConnection.timeout` property can be changed at this point.
 
-Then response headers (and other info) are read from the connection via `HTTPConnection.getresponse()` and returned as a `urllib3.HTTPResponse`. The `HTTPResponse` instance carries a reference to the `HTTPConnection` instance. (Within `HTTPConnectionPool` the `HTTPResponse._connection` is set to `response_conn`, is that needed or can `HTTPConnection.getresponse()` handle this?)
+```python
+conn.timeout = 5.0  # (read timeout)
+```
 
-If pooling is in use set `_connection` and `_pool` on the `HTTPResponse` instance. If retries are in use set `retries` on the `HTTPResponse` instance.
+Then the HTTP request can be sent with `HTTPConnection.request()`. If a `BrokenPipeError` (should we include `SSLEOFError`?) is raised while sending the request body it can be swallowed as a response can still be received from the origin even when the request isn't completely sent.
```

Comment:
Regarding your question, I'm not clear on whether you can read anything after `SSLEOFError` was raised. If we can read, then it's worth swallowing? Or is it something actionable where the urllib3 user should change something and retry?